contracts/Pair.sol

```diff
@@ -148,24 +176,46 @@ contract Pair is IPair {
         }
     }
 
-    // Accrue fees on token0
+    // Accrue fees on token0.
     function _update0(uint amount) internal {
-        _safeTransfer(token0, fees, amount); // transfer the fees out to PairFees
-        uint256 _ratio = amount * 1e18 / totalSupply; // 1e18 adjustment is removed during claim
-        if (_ratio > 0) {
-            index0 += _ratio;
+        if (hasGauge == false) {
+            _safeTransfer(token0, tank, amount); // transfer the fees to tank MSig for gaugeless LPs
+            uint _ratio = (amount * 1e18) / totalSupply; // 1e18 adjustment is removed during claim
+            if (_ratio > 0) {
+                index0 += _ratio;
+            }
+            emit TankFees(token0, amount, tank);
+        }
+        if (hasGauge == true) {
+            // _safeApprove(token0, externalBribe, amount);  // max abprove when setExternalBribe() is called
+            IBribe(externalBribe).notifyRewardAmount(token0, amount); //transfer fees to exBribes
+            //  _safeTransfer(token0, tank, amount);
+            uint _ratio = (amount * 1e18) / totalSupply; // 1e18 adjustment is removed during claim
+            if (_ratio > 0) {
+                index0 += _ratio;
+            }
+            emit GaugeFees(token0, amount, externalBribe);
         }
-        emit Fees(msg.sender, amount, 0);
     }
 
     // Accrue fees on token1
     function _update1(uint amount) internal {
-        _safeTransfer(token1, fees, amount);
-        uint256 _ratio = amount * 1e18 / totalSupply;
-        if (_ratio > 0) {
-            index1 += _ratio;
+        if (hasGauge == false) {
+            _safeTransfer(token1, tank, amount); // transfer the fees to tank MSig for gaugeless LPs
+            uint _ratio = (amount * 1e18) / totalSupply; // 1e18 adjustment is removed during claim
+            if (_ratio > 0) {
+                index0 += _ratio;
+            }
+            emit TankFees(token1, amount, tank);
+        }
+        if (hasGauge == true) {
+            IBribe(externalBribe).notifyRewardAmount(token1, amount); //transfer fees to exBribes
+            uint _ratio = (amount * 1e18) / totalSupply; // 1e18 adjustment is removed during claim
+            if (_ratio > 0) {
+                index0 += _ratio;
```

Comment:
idk if this whole index thing is still needed but just in case pls make it index1 first

Reply:
IMO.. we can remove all of index0 stuff

Reply:
yes let's remove them and also claimFees

Reply:
removed

Reply:
removed